tools/src/test/scala-3/scala/scalanative/NIRCompilerTest3.scala

```diff
@@ -36,15 +36,23 @@ class NIRCompilerTest3 extends AnyFlatSpec with Matchers with Inspectors {
         |""".stripMargin))
     }.getMessage should include("extern method foo needs result type")
   }
+  it should "report error for extern in top-level val definition" in {
+    intercept[CompilationFailedException] {
+      NIRCompiler(_.compile("""
+        |import scala.scalanative.unsafe.extern
+        |
+        |val foo: Int = extern
+        |""".stripMargin))
+    }.getMessage should include("extern` cannot be used in val definition")
+  }
```

Comment:
I found a workaround. Adding mutability check for transformValDef in nscplugin/src/main/scala-3/scala/scalanative/nscplugin/PrepNativeInterop.scala cleans strange link errors. (though this makes it impossible to detect extern val definition at genClassFields by `isExtern && !mutable` 😢)

I guess  providing `extern` information with `val` symbol generated something that is invalid at link time and those invalid symbols are not filtered out.

I think we can avoid generating invalid symbol in advance at PrepNativeInterop.scala, but it may be possible to filter out such invalid symbol while/after tree transformation. But, I have no idea how and where Scala Native in Scala 2.x removes extern val definition. 

```diff
  override def transformValDef(vd: ValDef)(using Context): Tree = {
    val enumsCtx = EnumerationsContext.get
    import enumsCtx._
    val sym = vd.symbol
    vd match {
      case ValDef(_, tpt, ScalaEnumValue.NoName(optIntParam)) =>
        val nrhs = scalaEnumValName(sym.owner.asClass, sym, optIntParam)
        cpy.ValDef(vd)(tpt = transformAllDeep(tpt), nrhs)

      case ValDef(_, tpt, ScalaEnumValue.NullName(optIntParam)) =>
        val nrhs = scalaEnumValName(sym.owner.asClass, sym, optIntParam)
        cpy.ValDef(vd)(tpt = transformAllDeep(tpt), nrhs)

      case _ =>
        // Set `@extern` annotation for top-level extern variables
        if (isTopLevelExtern(vd) &&
            !sym.hasAnnotation(defnNir.ExternClass) 
++         && vd.symbol.is(Flags.Mutable)) {
          sym.addAnnotation(defnNir.ExternClass)
          sym.setter.addAnnotation(defnNir.ExternClass)
        }

        vd
    }
  }
```

at 

https://github.com/scala-native/scala-native/blob/6b92eb57ce43cffa55bc555a6eb05ba11e4f5d80/nscplugin/src/main/scala-3/scala/scalanative/nscplugin/PrepNativeInterop.scala#L60

Reply:
Can I detect top-level extern variables after PrepNativeInterop phase and report error and skip generation of invalid extern val.